sql/src/main/java/io/crate/planner/node/dql/Collect.java

```diff
@@ -147,6 +148,19 @@ public int numOutputs() {
     }
 
     @Override
+    public boolean executesOnShard() {
+        List<Projection> projections = collectPhase.projections();
+        if (projections.isEmpty()) {
+            return collectPhase instanceof RoutedCollectPhase &&
```

Comment:
Is it better to add a property to the CollectPhase and avoid instanceof?

Reply:
I don't think it matters - instanceof is not *that* expensive.